### DISCUSSION TITLE
Parquet File Reader

### DISCUSSION THREAD

**Discussion by pcarey778** [nextgenhealthcare/connect#6189] [Q&A]
Files are coming in encoded. Is it possible to decode them in connect?

**Comment by jonbartels** [comment]
Most likely yes. When you say encoded do you mean base64 encoding or something else?

**Reply by pcarey778** [reply]
Does not seem to be base64. I just know that it is a binary file and not readable as it comes in to the channel.

**Reply by jonbartels** [reply]
Can you share a sample? Is "Parquet" a file format or a file server? Is it https://parquet.apache.org/ ?

The first thing to determine is if you should read the file natively in Mirth (like a CSV or text format) or if you need a tool like a Java library to call from Mirth to parse the file.

**Comment by rjkroll** [comment]
You can try using Powershell and then using Mirth to execute it via commandLine in a transformer step:


```
Install-Module PSParquet

Import-Module PSParquet

$data = Import-Parquet -Path "path/to/file.parquet"

$data
```


or Python
```
import pyarrow.parquet as pq

parquet_file = pq.ParquetFile('example.parquet')

metadata = parquet_file.metadata

print(f"Number of rows: {metadata.num_rows}")
print(f"Number of columns: {metadata.num_columns}")
print(f"Encoding: {metadata.schema.encoding}")

table = pq.read_table('example.parquet')

df = table.to_pandas()

print(df)
```

**Reply by jonbartels** [reply]
https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Execute%20Runtime%20Command is a community contribution to run shell scripts from Mirth.
